frontend/lib/src/dataframes/arrowFormatUtils.ts

```diff
@@ -304,33 +361,122 @@ function formatPeriod(duration: bigint, freq: PeriodFrequency): string {
+ */
+function formatObject(object: any, field?: Field): string {
+  if (field?.type instanceof Struct) {
+    // This type is used by python dictionary values
+
+    // Workaround: Arrow JS adds all properties from all cells
+    // as fields. When you convert to string, it will contain lots of fields with
+    // null values. To mitigate this, we filter out null values.
+
+    return JSON.stringify(object, (_key, value) => {
+      if (!notNullOrUndefined(value)) {
+        // Ignore null and undefined values ->
+        return undefined
+      }
+      if (typeof value === "bigint") {
+        return Number(value)
+      }
+      return value
+    })
+  }
+  return JSON.stringify(object, (_key, value) =>
+    typeof value === "bigint" ? Number(value) : value
+  )
+}
+
+/**
+ * Formats a float value to a string.
+ *
+ * @param num The float value to format.
+ * @returns The formatted float value.
+ */
+function formatFloat(num: number): string {
+  if (!Number.isFinite(num)) {
+    return String(num)
+  }
+
+  return numbro(num).format("0,0.0000")
+}
+
+/**
+ * Formats an interval value from arrow to string.
+ */
+function formatInterval(x: StructRow, field?: Field): string {
+  // Serialization for pandas.Interval is provided by Arrow extensions
   // https://github.com/pandas-dev/pandas/blob/235d9009b571c21b353ab215e1e675b1924ae55c/
   // pandas/core/arrays/arrow/extension_types.py#L17
-  const extensionName = field.metadata.get("ARROW:extension:name")
-  if (extensionName) {
+  const extensionName = field && field.metadata.get("ARROW:extension:name")
+  if (extensionName && extensionName === "pandas.interval") {
     const extensionMetadata = JSON.parse(
       field.metadata.get("ARROW:extension:metadata") as string
     )
-    if (extensionName === "pandas.interval") {
-      const { subtype, closed } = extensionMetadata
-      return formatInterval(x as StructRow, subtype, closed)
-    }
-    if (extensionName === "pandas.Period") {
-      const { freq } = extensionMetadata
-      return formatPeriod(x as bigint, freq)
-    }
+    const { subtype, closed } = extensionMetadata
+
+    const interval = (x as StructRow).toJSON() as Interval
+
+    const leftBracket = closed === "both" || closed === "left" ? "[" : "("
+    const rightBracket = closed === "both" || closed === "right" ? "]" : ")"
+
+    const leftInterval = format(
+      interval.left,
+      {
+        pandas_type: subtype,
+        numpy_type: subtype,
+      },
+      (field.type as Struct)?.children?.[0]
+    )
+    const rightInterval = format(
+      interval.right,
+      {
+        pandas_type: subtype,
+        numpy_type: subtype,
+      },
+      (field.type as Struct)?.children?.[1]
+    )
+
+    return `${leftBracket + leftInterval}, ${rightInterval + rightBracket}`
   }
   return String(x)
 }
 
-/** Takes the data and it's type and nicely formats it. */
+/** Takes the cell data and type metadata from arrow and nicely formats it. */
```

Comment:
nit: could you explain in a short paragraph what the difference between the `type` and the `field` is, like what information is provided in each of them. I think that would help (me) to understand the arrow format better and what is supposed to go in where.

Reply:
In short: field is what arrow refers to the column metadata. This contains a data type (e.g. number, string, ...) and some other metadata required to describe the type. Based on this field metadata and the pandas metadata that is also embedded in the arrow table, we are currently constructing our own type object for every column. But I plan to do some subsequent refactoring around the types in another PR and will try to make this a bit more understandable.

Reply:
Added a bit more docstring info, but the subsequent PR will simplify this

Reply:
Great, thank you!